### DISCUSSION TITLE
Case sensitive matching?

### DISCUSSION THREAD

**Discussion by Jackenmen** [community/community#9759] [Other Feature Feedback, Questions, & Ideas] [answered]
When using `grep.app`, I often find myself using the "Case sensitive" option to quickly restrict the matches to what I'm really looking for. In `cs.github.com`, there is no such option.
When I'm using this option, often it's about symbols for which I could use the symbol qualifier instead but I feel this could be useful in other cases.

**Comment by look** [comment]
This is on our roadmap. Stay tuned!

**Reply by yaelz** [reply]
Any news? 🤞

**Reply by look** [reply]
We are still planning on implementing this, but we're prioritizing scaling out the system first.

**Reply by lindhe** [reply]
Staying tuned... 👀

**Reply by luckynick** [reply]
Omg please just do the simple usable searching features lol)

**Reply by Neme12** [reply]
This is such a basic feature, along with "match whole world", that almost every search implementation like in browsers or editors supports. Why is this still missing? The search is completely unusable. 😔

**Reply by bounlu** [reply]
We are over-tuned already.

**Reply by davidtowoju** [reply]
Still tuned in, but the feature’s on mute. 🔇

**Reply by agiUnderground** [reply]
So 4 years is not enough? Any chance it would be implemented by 2030?

**Reply by mP1** [reply]
If only github used AI!

**Reply by daivid-drivemode** [reply]
@look still on the roadmap? Maybe this was implemented in some parts of GH, but not all? It doesn't work on Github Action's logs from builds at the moment. The response from @Jackenmen doesn't work there either.

**Comment by rogchap** [comment]
Would be also great if it could support Regex flags for example `/FOO/i` as Regex should default to case sensitive.

**Comment by balramr** [comment]
I wonder if "smart" case sensitivity could be an option, i.e. if the search is all lower case, it defaults to case-insensitive. But if your search is mixed-case, it does a case-sensitive search, as that was probably your intention.

**Reply by shric** [reply]
I prefer @rogchap's suggestion, which is uniform with regex syntax used elsewhere. I often want all lower case **and** case sensitive.

**Reply by mP1** [reply]
Your smart suggestion simply creates problems for the many users who want to use mixed case terms that are case sensitive.

Java for example has many examples of mixed case tokens, and your suggestion would be broken. Its simply better to have a checkbox or whatever in the UI to allow the user to pick case sensitivity/insensitivity.

**Comment by nexuspcs** [comment]
I would love this

**Comment by steveluscher** [comment]
Just to add some color, I typically _rely_ on case sensitive matches when searching for the thing I need. It helps me disambiguate between types and classes in Typescript (eg. `ConfirmTransaction`, `Transaction`) and functions and variables (eg. `confirmTransaction`, and `transaction`).

**Comment by maxkosty** [comment, marked as answer]
In the meantime this can be accomplished with `(?-i)`:

e.g. `/(?-i)myCaseSensitiveSearchString/`

**Reply by Jackenmen** [reply]
While not a resolution from GitHub, I think it's worth highlighting this workaround with the "mark as answer" until a better way is available.

**Reply by joweeba** [reply]
This workaround doesn't appear to work for Enterprise search

**Comment by swistak** [comment]
Years later still not solved. Who in the right mind would make code search case insensitive? Seriously.

Even regexp and "exact match" are case insensitive.

Who even allowed this to go into production?

**Comment by rking-bennie** [comment]
Would be nice to see search made better. This would level GitHub up.

**Comment by jontwo** [comment]
Still staying tuned. This is a code website, why doesn't it do case sensitive search? In my IDE I can easily do case sensitive and whole word searching, this isn't difficult.

**Comment by joweeba** [comment]
Your organization needs to prioritize this one. For larger customers, it's impossible to search large code bases accurately and efficiently using your product. Having this feature would have saved me hours this week.

**Comment by shtyelus** [comment]
This is pathetic!

**Comment by martin-honek-tg** [comment]
Four years on the roadmap. This is a very long roadmap they have in GitHub! :)

**Reply by mP1** [reply]
Maybe github should use AI to finish the feature in 5 minutes.

**Comment by SiefSeif** [comment]
Anyone really cares? it's a simple very important feature
